xql/main.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import xarray as xr
+import typing as t
+
+from sqlglot import parse_one, exp
+
+operate = {
+    "and" : lambda a, b: a & b,
+    "or" : lambda a, b: a | b,
+    "eq" : lambda a, b: a == b,
+    "gt" : lambda a, b: a > b,
+    "lt" : lambda a, b: a < b,
+    "gte" : lambda a, b: a >= b,
+    "lte" : lambda a, b: a <= b,
+}
+
+aggregate_function_map = {
+    'avg': lambda x, y: x.mean(dim=y) if y else x.mean(),
+    'min': lambda x, y: x.min(dim=y) if y else x.mean(),
```

Comment:
I think you meant `x.min()`.
Description of what's happening here is helpful. It's better to create a Factory class of this instead of creating a map, that way you can apply better validations to these and future functions.